test/transform.test.ts

```diff
@@ -108,6 +108,164 @@ describe('transform', function () {
       });
     });
 
+    context('if contains condition function', function () {
```

Comment:
👏  Excellent job with the test cases! I would add one more like [here](https://github.com/nikostoulas/metamorphosi/blob/d1318cc10c56e2e0210f83aabe83b47c8242a94c/test/transform.test.ts#L298).
A case where the condition function throws an error (it is user-provided so it could have errors) to see what's the outcome.  I would expect it to be treated as a function that returns undefined (As all evaluations that throw an error do [here](https://github.com/nikostoulas/metamorphosi/blob/3b5aa88e5c8d8e58478a3d5345c845075acfb825/src/eval.ts#L26). And possibly be consdired a falsy value and prune the object.

An user example could be 
```js
if:[($)=> $.a.b.c.d] 
```
where c is undefined so this would throw a null pointer exception. You would expect this to prune the object as $.a.b.c.d does not exist.